internal/database/users.go

```diff
@@ -675,6 +679,65 @@ func logUserDeletionEvents(ctx context.Context, db DB, ids []int32, name Securit
 	db.SecurityEventLogs().LogEventList(ctx, events)
 }
 
+// RecoverUserByID recovers a user by their ID.
+func (u *userStore) RecoverUserByID(ctx context.Context, id int32) error {
+	return u.RecoverList(ctx, []int32{id})
+}
+
+// RecoverList recovers a list of users by their IDs.
+func (u *userStore) RecoverList(ctx context.Context, ids []int32) error {
+	tx, err := u.Transact(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	userIDs := make([]*sqlf.Query, len(ids))
+	for i := range ids {
+		userIDs[i] = sqlf.Sprintf("%d", ids[i])
+	}
+	idsCond := sqlf.Join(userIDs, ",")
+	res, err := tx.Query(ctx, sqlf.Sprintf("UPDATE users SET deleted_at=NULL, updated_at=now() WHERE id IN (%s) AND deleted_at IS NOT NULL RETURNING id", idsCond))
+	if err != nil {
+		return err
+	}
+
+	affectedUserID := []int32{}
+	for res.Next() {
+		var id int32
+		if err := res.Scan(&id); err != nil {
+			return err
+		}
+		affectedUserID = append(affectedUserID, id)
+	}
```

Comment:
```suggestion
	affectedUserID, err := basestore.ScanInt32s(tx.Query(ctx, sqlf.Sprintf("UPDATE users SET deleted_at=NULL, updated_at=now() WHERE id IN (%s) AND deleted_at IS NOT NULL RETURNING id", idsCond)))
	if err != nil {
		return err
	}
```

That should work :) `basestore` has some great stuff.

Reply:
That is awesome 🔥 , Thank you 🙏